client/src/components/InstanceSummaryView.js

```diff
@@ -28,53 +34,70 @@ const InstanceSummaryView = () => {
     stringMinCredits, 
     stringMaxCredits, 
     gradingScale, 
-    teachers 
+    teachers,
+    startingPeriod,
+    endingPeriod 
   } = useOutletContext();
 
-  // state variables handling the alert messages
-  const [snackPack, setSnackPack] = useState([]);
-  const [alertOpen, setAlertOpen] = useState(false);
-  const [messageInfo, setMessageInfo] = useState(undefined);
+  const [created, setCreated] = useState(false);
 
-  // useEffect in charge of handling the back-to-back alerts
-  // makes the previous disappear before showing the new one
-  useEffect(() => {
-    if (snackPack.length && !messageInfo) {
-      setMessageInfo({ ...snackPack[0] });
-      setSnackPack((prev) => prev.slice(1));
-      setAlertOpen(true);
-    } else if (snackPack.length && messageInfo && alertOpen) {
-      setAlertOpen(false);
-    }
-  }, [snackPack, messageInfo, alertOpen]);
+  const [setInstanceAlert, messageInfo, setMessageInfo, alertOpen, setAlertOpen] = useSnackPackAlerts();
+  const [setAttainmentAlert, messageInfo2, setMessageInfo2, alertOpen2, setAlertOpen2] = useSnackPackAlerts();
 
   const onGoBack = () => {
     navigate('/' + courseId + '/add-attainments/' + sisuInstanceId);
   };
 
-  // Temporary to fake the effect of loading
+  // Helper function
   const sleep = ms => new Promise(r => setTimeout(r, ms));
 
   const onCreateInstance = async () => {
-    setSnackPack((prev) => [...prev, loadingMsg]);
 
-    // TODO: replace and actually create the instance, 
-    // make success dependant on the result 
-    await sleep(2000);
-    const success = true; 
-    await sleep(1000);
-    if (success) {
-      setSnackPack((prev) => [...prev, successMsg]);
-      await sleep(4000);
-      navigate('/course-view/' + courseId);
-    } else {
-      setSnackPack((prev) => [...prev, errorMsg]);
+    // attempt to create instance
+    try {
+      const instanceObj = { 
+        gradingScale: textFormatServices.convertToServerGradingScale(gradingScale),
+        sisuCourseInstanceId: sisuInstanceId,
+        type: courseType,
+        teachersInCharge: [1],                   // fake ! TODO: replace with teachers when figured out how to fetch ids (currently strings)
+        startingPeriod: startingPeriod ?? 'I',   // fake ! TODO: replace with just startingPeriod when it's not just null anymore
```

Comment:
A better comment here could be that 'Delete starting and ending periods after these are not requirement by the DB anymore', because they shouldn't be requirements. I will address this in the notes about Frontend Knowledge Transfer.

Reply:
Yep, I'll change these